hoot-services/src/main/java/hoot/services/jobs/JobsStatusesManagerImpl.java

```diff
@@ -137,21 +137,35 @@ public JobHistory getJobsHistory(Users user, String sort, long offset, long limi
                 .limit(limit)
                 .fetch();
 
-        List<JobStatusResponse> jobs = jobsHistory.stream().map(j -> {
-            JobStatusResponse response = new JobStatusResponse();
-            response.setJobId(j.getJobId());
-            response.setJobType(JobType.fromInteger(
-                    (j.getJobType() != null) ? j.getJobType() : JobType.UNKNOWN.ordinal()
-                ).toString());
-            response.setMapId(j.getResourceId());
-            response.setStart(j.getStart().getTime());
-            response.setEnd(j.getEnd().getTime());
-            response.setStatus(hoot.services.job.JobStatus.fromInteger(j.getStatus()).toString());
-            response.setStatusDetail(j.getStatusDetail());
-            response.setTags(PostgresUtils.postgresObjToHStore(j.getTags()));
+        List<String> familyIds = new ArrayList<>();
+        if (!groupJobId.equals("")) {
+            List<String> processedIds = new ArrayList<>();
+            findJobsFamily(familyIds, processedIds, groupJobId);
+        }
 
-            return response;
-        }).collect(Collectors.toList());
+        List<JobStatusResponse> jobs = jobsHistory.stream()
+            .filter(j -> {
+                if (!familyIds.isEmpty()) {
+                    return familyIds.contains(j.getJobId());
+                }
+
+                return true;
```

Comment:
if we do a filter on a job it will always have at least 1 job in the list, the target job.

The not empty is only for cases when there isnt a group job query since `familyIds` will never get populated